File: kotlin/linearSearch.kt
// 線形探索
fun main(args: Array<String>) {
    println(linearSearch(arrayOf(104,865,413,541,304,354,165,150), 304));
}

fun linearSearch(arr: Array<Int>, target: Int): Int {
    for(i in arr.indices){
        if(arr[i] == target) {
            return i;
        }
    }
    return -1;
}

File: csharp/linearSearch.cs
// 線形探索
using System.IO;
using System;

class Program {
    static void Main() {
        int[] arr = { 104,865,413,541,304,354,165,150 };
        Console.WriteLine(linearSearch(arr, 304));
    }
    static int linearSearch (int[] arr, int target) {
        for (int i = 0; i < arr.Length; i++) {
            if (arr[i] == target) {
                return i;
            }
        }
        return -1;
    }
}
